### PR TITLE
Support ``numeric_only=True`` in ``GroupBy.corr`` and ``GroupBy.cov``

### PR DESCRIPTION
**Pull request by phofl**
- [ ] Closes #xxxx
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

sits on top of #10222 as well

### CODE REVIEW

File: dask/dataframe/groupby.py [hunk 1]
```diff
@@ -2102,6 +2105,8 @@ def corr(self, ddof=1, split_every=None, split_out=1, numeric_only=no_default):
         """Groupby correlation:
         corr(X, Y) = cov(X, Y) / (std_x * std_y)
         """
+        if not PANDAS_GT_150 and numeric_only is not no_default:
```
Comment (j-bennet) [inline dask/dataframe/groupby.py:2108]: This part is used in several places, might want to move it into the decorator.

File: dask/dataframe/groupby.py [hunk 2]
```diff
@@ -347,7 +347,7 @@ def wrapper(self, *args, **kwargs):
                 if not PANDAS_GT_150 and (
                     numeric_only is False
                     or numeric_only is not no_default
-                    and func.__name__ in ["corr", "cov", "cumprod", "cumsum"]
+                    and func.__name__ in ["cumprod", "cumsum"]
```
Comment (j-bennet) [inline dask/dataframe/groupby.py:350]: `corr` and `cov` still don't support `numeric_only=False`, right? or do they now?

File: dask/dataframe/groupby.py [hunk 3]
```diff
@@ -2124,6 +2129,11 @@ def cov(
 
         When `std` is True calculate Correlation
         """
+        if not PANDAS_GT_150 and numeric_only is not no_default:
+            raise TypeError("numeric_only not supported for pandas < 1.5")
+        numeric_only_kwargs = (
```
[the remark this answers is not in the document]
Comment (phofl) [reply dask/dataframe/groupby.py:2134]: Not yet

File: dask/dataframe/groupby.py (hunk 1 above)
Comment (phofl) [reply dask/dataframe/groupby.py:2108]: The decorator is too complicated right now imo. I’d prefer keeping this here and refactoring when I am through

File: dask/dataframe/groupby.py (hunk 2 above)
Comment (phofl) [reply dask/dataframe/groupby.py:350]: No, this is about true only

Comment (j-bennet) [reply dask/dataframe/groupby.py:350]: So then why are you removing `corr` and `cov` from this clause?

Comment (phofl) [reply dask/dataframe/groupby.py:350]: False would error a few lines before this clause. This condition is only about true

**Comment by jrbourbeau** [comment]
Merging https://github.com/dask/dask/pull/10224 introduced some merge conflicts here -- went ahead and resolved them

**Comment by phofl** [comment]
Thx, pushed a small update.

**Review by jrbourbeau — APPROVED** [review]

**Pull request merged**